### PR TITLE
Document ClipboardData constructor breaking change

### PR DESCRIPTION
**Pull request by loic-sharma**
Document the breaking change introduced by https://github.com/flutter/flutter/pull/122446.

I will create a dart fix to automate this migration.

Part of: https://github.com/flutter/flutter/issues/121976

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/release/breaking-changes/clipboard-data-required.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+The [`ClipboardData constructor`][]'s `text` argument is no longer nullable.
+Code that provides `null` to the `text` argument must be migrated to provide
+an empty string `''`.
+
+## Context
+
+In preparation for supporting multiple clipboard data variants, the
+`ClipboardData` constructor now requires that at least one data variant is
+provided.
```
Comment (loic-sharma) [inline src/release/breaking-changes/clipboard-data-required.md:16]: I didn't mention this, but platforms were inconsistent in how they handled `null`. Some would reset the clipboard, some set the clipboard to the text `"null"`, others would crash. This was an unintended consequence of platforms assuming that at least one data variant is provided. We made the behavior consistent by disallowing this case.

Comment (sfshaza2) [reply src/release/breaking-changes/clipboard-data-required.md:16]: How about something like this:

"Previously, platforms were inconsistent in how they handled `null`. Behavior is now consistent across platforms. If you are interested in the low level details, see PR xxx."

I can imagine that some would want more info.

**Review by sfshaza2 — APPROVED** [review]
This lgtm. I won't merge this in case you want to add a note about the inconsistent behavior, but this looks good otherwise!

File: src/release/breaking-changes/clipboard-data-required.md (hunk 1 above)
Comment (loic-sharma) [reply src/release/breaking-changes/clipboard-data-required.md:16]: Added!

**Pull request merged**